src/ansi-c/ansi_c_internal_additions.cpp

```diff
@@ -120,6 +122,25 @@ static std::string architecture_string(T value, const char *s)
          std::string(s) + "=" + std::to_string(value) + ";\n";
 }
 
+using max_alloc_sizet = uint64_t;
```

Comment:
Shouldn't this be `size_t`? That's the type with which `__CPROVER_max_malloc_size` is declared.

Reply:
Yes this makes sense!

Reply:
Though the advantage of using a platform agnostic width is writing the unit tests is easier

Reply:
As pointed out by @thk123 in private communication, when using `size_t` instead of `uint64_t`, the computation would overflow when running on a 32-bit machine and using e.g. `--LLP64`.

Reply:
Could we please use `mp_integer` here instead of any architecturally limited type? We do not want to run into limitations of the host or compiler with this computation.